File: ChatAppAngular/src/app/user/payload/conversation-message-history-response.ts

export class ConversationMessageHistoryResponse {
  id: number=0;
  sender: number=0;
  recipient: number=0;
  message: string='';
  conversationId: string='';
  sentAt: string='';
  recieved: string='';
  chatFiles: any[]=[]; // Assuming ChatFiles is another class
  images: any[]=[]; // Assuming ChatFiles is another class
  isSeen: string='';

  constructor(
    id: number,
    sender: number,
    recipient: number,
    message: string,
    conversationId: string,
    sentAt: string,
    recieved: string,
    chatFiles: any[],
    images: any[],
    isSeen: string
  ) {
    this.id = id;
    this.sender = sender;
    this.recipient = recipient;
    this.message = message;
    this.conversationId = conversationId;
    this.sentAt = sentAt;
    this.recieved = recieved;
    this.chatFiles = chatFiles;
    this.images = images;
    this.isSeen = isSeen;
  }
}

File: ChatAppBackend/src/main/java/com/gapshap/app/payload/ConversationMessageHistoryResponse.java
package com.gapshap.app.payload;

import java.util.List;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.gapshap.app.model.chat.ChatFiles;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class ConversationMessageHistoryResponse {

	private Long id;
	
	private Long sender;
	private Long recipient;
	
	   private String message;
	
	private String conversationId;

	private String sentAt;

//	private String recieved;
	@JsonIgnoreProperties(value= {"message"})
	private List<ChatFiles> chatFiles;
	
	@JsonIgnoreProperties(value="message")
	private List<ChatFiles> images;
	
	private String isSeen;

}
